x-pack/plugins/ml/public/application/explorer/explorer_charts/explorer_chart_single_metric.js

```diff
@@ -448,6 +471,37 @@ export class ExplorerChartSingleMetric extends React.Component {
         .attr('y', (d) => lineChartYScale(d.value) - SCHEDULED_EVENT_SYMBOL_HEIGHT / 2);
     }
 
+    function showAnomalyPopover(marker, circle) {
+      const anomalyTime = marker.date;
+
+      // The table items could be aggregated, so we have to find the item
+      // that has the closest timestamp to the selected anomaly from the chart.
+      const tableItem = that.props.tableData.anomalies.reduce((closestItem, currentItem) => {
```

Comment:
Looks like this is the same logic used in `explorer_chart_distribution.js` - might be worth moving into a shared function with unit tests.

Reply:
👍 moved this to a util function including unit tests in [5ac1cbb](https://github.com/elastic/kibana/pull/186587/commits/5ac1cbbd8208f2ff7c8c519cca40c950083d0e4e).